servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultHttpLoadBalancerFactory.java

```diff
@@ -171,7 +309,41 @@ public <T> Publisher<? extends T> transportEventStream(final HttpEventKey<T> eve
 
         @Override
         public Single<StreamingHttpResponse> request(final StreamingHttpRequest request) {
-            return delegate.request(request);
+            return Single.defer(() -> {
+                if (tracker == null) {
+                    return delegate.request(request).shareContextOnSubscribe();
+                }
+
+                final RequestTracker theTracker = new AtMostOnceDeliveryRequestTracker(tracker);
+                final long startTime = theTracker.beforeStart();
+
+                return delegate.request(request)
+                        .map(response -> {
+                            final ErrorClass eClass = peerResponseErrorClassifier.apply(response);
+                            if (eClass != null) {
+                                // The onError is triggered before the body is actually consumed.
+                                theTracker.onError(startTime, eClass);
+                            }
+                            return response;
+                        })
+                        .liftSync(new BeforeFinallyHttpOperator(new TerminalSignalConsumer() {
```

Comment:
If you use `BeforeFinallyHttpOperator(TerminalSignalConsumer, true)`, there is no need to wrap tracker with `AtMostOnceDeliveryRequestTracker`

Reply:
The `AtMostOnceDeliveryRequestTracker` is still needed for the `map` the successful response -> error use-case.